### PR TITLE
rewrite card positioning using media queries

### PR DESCRIPTION
**Pull request by serentinos**
- [DEMO LINK](https://serentinos.github.io/layout_catalog_grid/)

### CODE REVIEW

File: src/styles/_mixin.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+@mixin responsive ($map, $width) {
+  @each $size, $columns in $map {
+    @media (max-width: $size) {
+      grid-template-columns: repeat($columns, $width);
+    }
+  }
+}
```
Comment (annholovko) [inline src/styles/_mixin.scss]: it's amazing!!! now I also want to do in such way😁🥺

File: src/styles/_catalog.scss [hunk 2]
```diff
@@ -1,22 +1,21 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: repeat(auto-fill, $card-width);
+  justify-content: center;
+  gap: $card-gap;
 
   max-width: 944px;
```
Comment (annholovko) [inline src/styles/_catalog.scss]: Once Ihor told us that it's better to start with mobile media-query

**Review by annholovko — APPROVED** [review]
Good job!🚀

File: src/styles/_catalog.scss (hunk 2 above)
Comment (serentinos) [reply src/styles/_catalog.scss]: Oh, I decided to use desktop-first in this task:)

File: src/styles/_mixin.scss (hunk 1 above)
Comment (serentinos) [reply src/styles/_mixin.scss]: Thank you!I appreciate that 🥰

Comment (Ihor-Karpyn) [inline src/styles/_mixin.scss]: The use of this loop worsens the readability of the code. In order to fully comprehend what is happening, one must check the lot of variables

Remove it, and remove the mixin that you use once.

File: src/styles/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
+$media-desktop: 1024px;
+$media-tablet: 768px;
+$media-mobile: 488px;
+
+$media-columns: (
+  $media-desktop : 3,
+  $media-tablet: 2,
+  $media-mobile : 1
+);
```
Comment (Ihor-Karpyn) [inline src/styles/_variables.scss]: Something strang
```suggestion
```

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

File: src/styles/_variables.scss (hunk 3 above)
Comment (serentinos) [reply src/styles/_variables.scss]: This is for media loop :D

**Review by Ihor-Karpyn — APPROVED** [review]
GJ

**Pull request closed without merging**